### PR TITLE
fix(content): check permissions for both nav items and content pages

### PR DESCRIPTION
**Pull request by bertyhell**
closes: https://district01.atlassian.net/browse/AVO2-950

* Allows an avo admin to select which nav items should be visible
    * A warning is show if he selects a user group for which the selected content page isn't accessable
* Allows an avo admin to select which users can visit the content page
* Store content pages by id in nav items, instead of by path. So we can 

image

image

### CODE REVIEW

File: src/workspace/views/Workspace.tsx [hunk 1]
```diff
@@ -50,20 +50,26 @@ export interface WorkspaceProps extends DefaultSecureRouteProps<{ tabId: string
...
 
 	// Methods
+	// react to route changes by navigating back wih the browser history back button
+	useEffect(() => {
+		setTabId(match.params.tabId);
+	}, [match.params.tabId]);
+
 	// Make map for available tab views
 	useEffect(() => {
+		console.log('init tabs', tabCounts, permissions, t, history, location, match, user);
```
Comment (benjaminnaesen) [inline src/workspace/views/Workspace.tsx]: Iew, a `console.log`! 😨

File: src/workspace/views/Workspace.tsx [hunk 2]
```diff
@@ -111,30 +117,28 @@ const Workspace: FunctionComponent<WorkspaceProps> = ({ history, match, user, ..
...
 
-	const getTabId = useCallback(() => {
-		return tabId || Object.keys(tabs)[0];
-	}, [tabs, tabId]);
-
 	// Get active tab based on above map with tabId
 	const getActiveTab = useCallback(() => {
-		return tabs[getTabId()];
-	}, [tabs, getTabId]);
+		console.log('get active tab');
```
Comment (benjaminnaesen) [inline src/workspace/views/Workspace.tsx]: Stab the `console.log`. 🔪

File: src/workspace/views/Workspace.tsx [hunk 3]
```diff
@@ -111,30 +117,28 @@ const Workspace: FunctionComponent<WorkspaceProps> = ({ history, match, user, ..
...
+		return tabs[tabId || Object.keys(tabs)[0]];
+	}, [tabs, tabId]);
 
 	useEffect(() => {
+		console.log('fetch permissions');
```
Comment (benjaminnaesen) [inline src/workspace/views/Workspace.tsx]: Opportunity to roadkill a `console.log`. 🚗

File: src/workspace/views/Workspace.tsx [hunk 4]
```diff
@@ -175,15 +179,11 @@ const Workspace: FunctionComponent<WorkspaceProps> = ({ history, match, user, ..
...
+	}, [user, t, setPermissions, permissions]);
 
 	useEffect(() => {
-		if (
-			!isEmpty(permissions) &&
-			!isEmpty(tabs) &&
-			loadingInfo.state !== 'loaded' &&
-			loadingInfo.state !== 'error'
-		) {
+		console.log('set loaded or error');
```
Comment (benjaminnaesen) [inline src/workspace/views/Workspace.tsx]: This `console.log` scared me. 👻

File: src/workspace/views/Workspace.tsx [hunk 5]
```diff
@@ -201,13 +201,14 @@ const Workspace: FunctionComponent<WorkspaceProps> = ({ history, match, user, ..
 		}
 	}, [setLoadingInfo, getActiveTab, t, permissions, tabs]);
 
-	const getNavTabs = () => {
+	const getNavTabs = useCallback(() => {
+		console.log('get nav tabs');
```
Comment (benjaminnaesen) [inline src/workspace/views/Workspace.tsx]: Crash a plane into this `console.log`. ✈️

File: src/admin/menu/views/MenuEdit.tsx [hunk 6]
```diff
@@ -104,6 +127,111 @@ const MenuEdit: FunctionComponent<MenuEditProps> = ({ history, match }) => {
...
+							userGroup => userGroup.value === faultyUserGroupId
+						);
+						return faultyUserGroup ? faultyUserGroup.label : null;
+					})
+				);
+				setPermissionWarning(
+					<div>
+						<Spacer margin="bottom-small">
+							<Trans>
+								Het navigatie item zal zichtbaar zijn voor gebruikers die geen toegang habben tot de
```
Comment (benjaminnaesen) [inline src/admin/menu/views/MenuEdit.tsx]: hebben*

File: src/admin/menu/views/MenuEdit.tsx [hunk 7]
```diff
@@ -104,6 +127,111 @@ const MenuEdit: FunctionComponent<MenuEditProps> = ({ history, match }) => {
...
+			}
+
+			const faultyUserGroupIds = without(navItemUserGroupIds, ...contentUserGroupIds);
+			if (faultyUserGroupIds.length) {
+				const faultyUserGroups = compact(
+					faultyUserGroupIds.map(faultyUserGroupId => {
+						const faultyUserGroup = allUserGroups.find(
+							userGroup => userGroup.value === faultyUserGroupId
+						);
+						return faultyUserGroup ? faultyUserGroup.label : null;
```
Comment (benjaminnaesen) [inline src/admin/menu/views/MenuEdit.tsx]: `return get(faultyUserGroup, 'label', null);` lijkt me mooier.

Comment (bertyhell) [reply src/admin/menu/views/MenuEdit.tsx]: true, maar minder efficiënt
ik heb het aangepast

**Pull request merged**